### PR TITLE
Blog : Creating Test Controllers Specs

### PR DESCRIPTION
**Pull request by aima98**
### For this milestone, I implemented the following features:

- [ ] Adding RSpec as a gem to my project.
- [ ] Creating a Request spec file for Users and Posts controllers.
- [ ] Writing test units for Users and Posts controllers to :
          1. check if the response status was correct.
          2. check if a correct template was rendered.
          3. check if the response body includes correct placeholder text.

### CODE REVIEW

File: spec/requests/users_spec.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+      get user_path(user)
+      expect(response).to render_template(:show)
+    end
+
+    it 'includes correct placeholder text in the response body' do
+      get user_path(user)
+      expect(response.body).to include('User Details')
+    end
+  end
+end
```
Comment (FrederickMih) [inline spec/requests/users_spec.rb:39]: - [ ] Hi @aima98, your tests are not running successfully caused by the duplicate user references in the likes table as you can see on the screenshot below:
Screenshot from 2023-09-26 12-19-11

**Review by FrederickMih — CHANGES_REQUESTED** [review]
## Hi :raising_hand: :raising_hand: @aima98, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!





## Highlights :green_circle: 

- `index` and `show` actions for `User` and `Post` controllers have been added as expected :+1: 


## Required Changes ♻️
**N/A**
_Check the comments under the review._

## Optional suggestions
**N/A**
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: db/schema.rb [hunk 2]
```diff
@@ -19,7 +19,6 @@
     t.datetime "created_at", null: false
     t.datetime "updated_at", null: false
     t.bigint "user_id", null: false
-    t.bigint "author_id"
     t.bigint "post_id", null: false
     t.index ["author_id"], name: "index_likes_on_author_id"
     t.index ["post_id"], name: "index_likes_on_post_id"
```
Comment (mckent05) [inline db/schema.rb:24]: **OPTIONAL** - I see that you edited your schema file, please note that it is not advisable to manually edit your schema file. The best practice is to create a migration that'd help you edit your schema. Please note this 👍

**Review by mckent05 — APPROVED** [review]
Hi @aima98, 

# STATUS: APPROVED ✔️ ✔️ 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations and well done! 🎉 
approved


## Highlights
- All linter checks are passing ✔️ ✔️ 
- Your PR is professional 👏 👏 
- You have done well creating tests for your controllers 👍 👍 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**